### PR TITLE
NU1605 fix for .net core clients

### PR DESCRIPTION
**Pull request by skomis-mm**
- fixes [#196](https://github.com/serilog/serilog-settings-configuration/issues/196)

Notes: the issue occurs when `RuntimeIdentifier` specified on publish/restore for `netstandard2.x` project with `Microsoft.NETCore.Targets=1.1.x` resolved by nuget dependency resolution. This change force to use `3.0.0` as suggested in [docs](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1605#example-2) and [related](https://github.com/dotnet/runtime/issues/25664) issue.

Cosmetic adjustments:
- fixed failing tests for net452 in VS TestExplorer
- added source link support
- for .NET Framework targets added reference assemblies packages and adjusted build script to be able to build projects under Linux environment

### CODE REVIEW

**Comment by nblumhardt** [comment]
Nice find! Looks like the fix, fingers crossed.

File: src/Serilog.Settings.Configuration/Serilog.Settings.Configuration.csproj [hunk 1]
```diff
@@ -15,23 +15,30 @@
     <PackageId>Serilog.Settings.Configuration</PackageId>
     <PackageTags>serilog;json</PackageTags>
     <PackageIcon>icon.png</PackageIcon>
-    <PackageProjectUrl>https://github.com/serilog/serilog-settings-configuration</PackageProjectUrl>
+    <PackageProjectUrl>https://serilog.net/</PackageProjectUrl>
```
Comment (tsimbalar) [inline src/Serilog.Settings.Configuration/Serilog.Settings.Configuration.csproj:18]: nit : this changes the link that the Nuget page points to  : 

https://www.nuget.org/packages/Serilog.Settings.Configuration/3.1.1-dev-00228

image

Not sure if it's a big deal, but if people is looking for docs specific to that package it becomes a little bit harder to find, what do you thing ?

Comment (skomis-mm) [reply src/Serilog.Settings.Configuration/Serilog.Settings.Configuration.csproj:18]: @tsimbalar , ah, right, a bad copy/paste from Serilog repository, will fix in next change

Comment (nblumhardt) [reply src/Serilog.Settings.Configuration/Serilog.Settings.Configuration.csproj:18]: I've been going on the assumption that "source repository" is the GitHub link, while "project" is more informational, but either way seems fine to me :-)

Comment (skomis-mm) [reply src/Serilog.Settings.Configuration/Serilog.Settings.Configuration.csproj:18]: @nblumhardt you are right. But unfortunately "source repository" is not shown in VS nor in the linqpad 😄 :
image

image

**Pull request merged**